### ISSUE TITLE
Will you release a pytorch version?

### ISSUE THREAD

**Issue by shaofengzeng** [rpautrat/SuperPoint#127]
Tensorflow is really hard, so could you give a pytorch version? Thanks

**Comment by rpautrat** [comment]
I might release a Pytorch version later, because I am now using it in Pytorch, but this is not a priority for now. If it happens, it would be in a few months more probably.

In the meantime, you can at least use the pretrained model of Magic Leap here: https://github.com/MagicLeapResearch/SuperPointPretrainedNetwork

**Comment by shaofengzeng** [comment]
> I might release a Pytorch version later, because I am now using it in Pytorch, but this is not a priority for now. If it happens, it would be in a few months more probably.
> 
> In the meantime, you can at least use the pretrained model of Magic Leap here: https://github.com/MagicLeapResearch/SuperPointPretrainedNetwork

Thanks, really looking forword to your pytorch version!

**Comment by rpautrat** [comment]
There is actually already a Pytorch version of SuperPoint here: [https://github.com/eric-yyjau/pytorch-superpoint](https://github.com/eric-yyjau/pytorch-superpoint), also taking inspiration from my repo. So it might just be easier to use their Pytorch repo if you need it in Pytorch.

**Comment by lwzbuaa** [comment]
@rpautrat It's greatfully for the pretrained model of superpoint model.I want to know whether the model can be used to multi-batch process the feature detection on the same time?how to use to multi-batch process the input images?

**Comment by Skydes** [comment]
Our recent [SuperGlue](https://github.com/magicleap/SuperGluePretrainedNetwork) paper has a PyTorch implementation of SuperPoint that can handle batches with multiple images: [superpoint.py](https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/models/superpoint.py).

**Comment by rpautrat** [comment]
There is currently no way to parallelize the feature detection across several batches at the same time in this repo. You can follow the suggestion of @Skydes if you need it in Pytorch, or write yourself the parallel loop if you prefer using this repo.

**Comment by lwzbuaa** [comment]
> Our recent [SuperGlue](https://github.com/magicleap/SuperGluePretrainedNetwork) paper has a PyTorch implementation of SuperPoint that can handle batches with multiple images: [superpoint.py](https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/models/superpoint.py).

@Skydes SuperGlue implement parallize very good，I want to know the model has some difference with the @rpautrat 's superpoint model?

**Comment by Skydes** [comment]
The SuperGlue repos is based on the orignal SuperPoint implementation.

**Comment by lwzbuaa** [comment]
> Our recent [SuperGlue](https://github.com/magicleap/SuperGluePretrainedNetwork) paper has a PyTorch implementation of SuperPoint that can handle batches with multiple images: [superpoint.py](https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/models/superpoint.py).

@Skydes in you project,the superpoint.py can use the gpu multibatch to parrel process multiimage at the same time?

**Comment by areslp** [comment]
> Our recent [SuperGlue](https://github.com/magicleap/SuperGluePretrainedNetwork) paper has a PyTorch implementation of SuperPoint that can handle batches with multiple images: [superpoint.py](https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/models/superpoint.py).

Hi, @Skydes, I have tried the one in superglue repo. I give two tensors to the superpoint network and measure the inference time. One tensor is `1 x 1 x h x w`, the other one is `batch x 1 x h x w`. I run the network in conda enviroment (pytorch 1.5, cuda 10.2) with 1080ti (10G). Batch size is set to 10.

Inference time for `1 x 1 x h x w` is about 0.01s.
Inference time for `batch x 1 x h x w` is about 0.1s.

It seems that the batch is not working..did I do something wrong?

**Comment by Skydes** [comment]
Even if the computations are batched, the inference time might not be constant, since your GPU might not have enough kernels to do all the computations in parallel. You might see better parallelism with smaller images or batch sizes.

**Comment by shaofengzeng** [comment]
Here is a pytorch version that has implemented most of the functions
[https://github.com/shaofengzeng/MySuperPoint](url)
